services/youtube/youtube-channel-base.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService, NotFound } = require('..')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const documentation = `
+<p>By using the YouTube badges provided by Shields.io, you are agreeing to be bound by the YouTube Terms of Service. These can be found here:
+<code>https://www.youtube.com/t/terms</code></p>`
+
+const schema = Joi.object({
+  pageInfo: Joi.object({
+    totalResults: nonNegativeInteger,
+    resultsPerPage: nonNegativeInteger,
+  }).required(),
+  items: Joi.array().items(
+    Joi.object({
+      statistics: Joi.object({
+        viewCount: nonNegativeInteger,
+        subscriberCount: nonNegativeInteger,
+        hiddenSubscriberCount: Joi.boolean().required(),
+        videoCount: nonNegativeInteger,
+      }).required(),
+    })
+  ),
+}).required()
+
+class YouTubeChannelBase extends BaseJsonService {
```

Comment:
As far as I can tell, this file is almost identical to `YouTubeVideoBase`. Could this please be refactored back to a single `YouTube` base? Some of the few differences between the two classes can be phased out:
* using the hostname, `www.googleapis.com`, as we've hitting the same underlying API in both cases.
* using the same check for the not found case. `if (!json.items || json.items.length === 0)` should cover both the video and channel responses.

The only remaining differences will be a few string values and the API schema - if you pass `{ videoId, channelId }` to the `handle` method, you should be able to infer the right values depending on which value is actually defined.